modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -1399,6 +1434,58 @@ void GDScriptAnalyzer::resolve_class_body(GDScriptParser::ClassNode *p_class, co
 		resolve_pending_lambda_bodies();
 	}
 
+	// Resolve base abstract class/method implementation requirements.
+	if (!p_class->is_abstract) {
+		HashSet<StringName> implemented_funcs;
+		int num_unimplemented_funcs = 0;
+		int num_abstract_funcs = 0;
+		String recent_method_signature;
+		const GDScriptParser::ClassNode *base_class = p_class;
+		while (base_class != nullptr) {
+			if (!base_class->is_abstract && base_class != p_class) {
+				break;
+			}
+			for (GDScriptParser::ClassNode::Member member : base_class->members) {
+				if (member.type == GDScriptParser::ClassNode::Member::FUNCTION) {
+					if (member.function->is_abstract) {
+						if (base_class == p_class) {
+							num_abstract_funcs++;
+						} else if (!implemented_funcs.has(member.function->identifier->name)) {
+							GDScriptParser::DataType base_type = base_class->datatype;
```

Comment:
```suggestion
							GDScriptParser::DataType base_class_type = base_class->datatype;
```

Declaration of `base_type` hides previous local declaration.